### PR TITLE
Adding a basic test for RazorGenerate

### PR DESCRIPTION
**Pull request by rynowak**
Lotsa cleanup

### CODE REVIEW

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs [hunk 1]
```diff
@@ -10,40 +13,88 @@ internal class Assert : Xunit.Assert
...
+            }
+
+            if (searchPattern == null)
+            {
+                throw new ArgumentNullException(nameof(searchPattern));
+            }
+
+            directoryPath = Path.Combine(result.Project.DirectoryPath, directoryPath);
+
+            var files = Directory.EnumerateFiles(directoryPath, searchPattern, SearchOption.AllDirectories).ToArray();
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs]: `Directory.GetFiles`

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs [hunk 2]
```diff
@@ -2,55 +2,49 @@
...
 
+#if DEBUG
+        protected string Configuration => "Debug";
+#elif RELEASE
+        protected string Configuration => "Release";
+#else
+#error Configuration not supported
+#endif
+
+        protected string IntermediateOutputPath => Path.Combine("obj", Configuration, TargetFramework);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs]: Nice

File: test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs [hunk 3]
```diff
@@ -10,40 +13,88 @@ internal class Assert : Xunit.Assert
...
+        {
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            var filePath = Path.Combine(result.Project.DirectoryPath, Path.Combine(paths));
+            if (File.Exists(filePath))
+            {
+                throw new FileMissingException(result, filePath);
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs]: This seems incorrect.

**Comment by rynowak** [comment]
@pranavkm updated with the changes to use `Reference`

File: test/testapps/SimpleMvc/SimpleMvc.csproj [hunk 4]
```diff
@@ -3,10 +3,16 @@
...
     <ProjectReference Include="$(OriginalProjectRoot)\..\..\Microsoft.AspNetCore.Razor.Test.MvcShim\Microsoft.AspNetCore.Razor.Test.MvcShim.csproj"/>
     <ProjectReference Include="$(OriginalProjectRoot)\..\..\..\src\Microsoft.AspNetCore.Razor.Runtime\Microsoft.AspNetCore.Razor.Runtime.csproj"/>
   </ItemGroup>
 
+  <ItemGroup Condition="'$(BinariesRoot)'!=''">
+    <Reference Include="$(BinariesRoot)\System.Diagnostics.DiagnosticSource.dll"/>
+    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Html.Abstractions.dll"/>
+    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.dll"/>
+    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Test.MvcShim.dll"/>
+  </ItemGroup>
```
Comment (rynowak) [inline test/testapps/SimpleMvc/SimpleMvc.csproj]: I don't want to get the deps file involved. The set of references we need is small and managable.

**Pull request merged**